### PR TITLE
feat: [PL-28262]: fix the issue of error while logging the exception

### PR DESCRIPTION
**Pull request by bhavya181**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37586)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37586)

**Review by adivishy1 — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37586)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcService.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public void createFile(CreateFileRequest request, StreamObserver<CreateFileRespo
         createFileResponse = harnessToGitHelperService.createFile(request);
         log.info(String.format("%s createFile ops response : %s", GIT_SERVICE, createFileResponse));
       } catch (Exception ex) {
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcService.java:149]: this is very repetitive and could have been moved to a method which takes enum as input.

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
changes req

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37586)* status: 3 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcService.java` line 149 at r1](https://reviewable.io/reviews/harness/harness-core/37586#-NC06JYzBifO5TeY_eg5-r1-149:-NCK_ufM2he3zWUVh8R-:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/36cc22ae7dc9896223317cc27b29d6bdc071ddca/136-git-sync-manager/src/main/java/io/harness/gitsync/sdk/HarnessToGitPushInfoGrpcService.java#L149)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

this is very repetitive and could have been moved to a method which takes enum as input.
</blockquote></details>

Done.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37586)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 2 of 2 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37586)* status: :shipit: complete! ✓

**Pull request merged**